src/utils/utils.ts

```diff
@@ -206,45 +206,33 @@ export module Utils {
     }
 
     /**
-     * Computes the max value from the array.
-     *
-     * If type is not comparable then t will be converted to a comparable before computing max.
+     * Applies the accessor, if provided, to each element of `array` and returns the maximum value.
+     * If no maximum value can be computed, returns defaultValue.
      */
-    export function max<C>(arr: C[], default_val: C): C;
-    export function max<T, C>(arr: T[], acc: (x?: T, i?: number) => C, default_val: C): C;
-    export function max(arr: any[], one: any, two?: any): any {
-      if (arr.length === 0) {
-        if (typeof(one) !== "function") {
-          return one;
-        } else {
-          return two;
-        }
-      }
+    export function max<C>(array: C[], defaultValue: C): C;
+    export function max<T, C>(array: T[], accessor: (t?: T, i?: number) => C, defaultValue: C): C;
+    export function max(array: any[], firstArg: any, secondArg?: any): any {
+      var accessor = typeof(firstArg) === "function" ? firstArg : null;
+      var defaultValue = accessor == null ? firstArg : secondArg;
       /* tslint:disable:ban */
-      var acc = typeof(one) === "function" ? one : typeof(two) === "function" ? two : undefined;
-      return acc === undefined ? d3.max(arr) : d3.max(arr, acc);
+      var maxValue = accessor == null ? d3.max(array) : d3.max(array, accessor);
       /* tslint:enable:ban */
+      return maxValue !== undefined ? maxValue : defaultValue;
     }
```

Comment:
Side note I think we will want to rewrite this so that defaultValue is always the second argument.  I think it's becoming a smell if we have a signature that says firstArg/secondArg and we have to decipher which is which  like the structure we have now.